test-runner/src/main/scala/com/typesafe/spark/test/mesos/framework/runners/DCOSClusterModeRunner.scala

```diff
@@ -0,0 +1,61 @@
+package com.typesafe.spark.test.mesos.framework.runners
+
+import Utils._
+import DCOSUtils._
+import com.amazonaws.auth.BasicAWSCredentials
+import com.amazonaws.services.s3.AmazonS3Client
+import com.amazonaws.services.s3.model.{CannedAccessControlList, ObjectMetadata, PutObjectRequest}
+import com.typesafe.config.Config
+import java.io.{FileInputStream, File}
+
+import org.apache.commons.io.FilenameUtils
+;
+
+object DCOSClusterModeRunner {
+  def run(dcosURL : String, applicationJarPath : String)(implicit config: Config): String = {
+    uploadToS3(applicationJarPath)
+
+    val basename = FilenameUtils.getBaseName(applicationJarPath)
+    val extension = FilenameUtils.getExtension(applicationJarPath)
+    val S3_BUCKET = sys.env("S3_BUCKET")
+    val S3_PREFIX = sys.env("S3_PREFIX")
+
+    val s3URL = s"http://${S3_BUCKET}.s3.amazonaws.com/${S3_PREFIX}${basename}.${extension}"
+
+    printMsg("Submitting spark tests...")
+    val submissionId = submitSparkJobDCOS(s3URL)
+
+    printMsg(s"Waiting for spark tests with task ID ${submissionId} to complete...")
+    waitForSparkJobDCOS(submissionId)
+
+    printMsg("Getting spark test output...")
+    getLogOutputDCOS(submissionId)
+  }
+
+  private def uploadToS3(path : String): Unit = {
+    printMsg("Uploading JAR to S3...")
+    val inputStream = new FileInputStream(path)
+    val basename = FilenameUtils.getBaseName(path)
+    val extension = FilenameUtils.getExtension(path)
+
+    val AWS_ACCESS_KEY = sys.env("AWS_ACCESS_KEY")
+    val AWS_SECRET_KEY = sys.env("AWS_SECRET_KEY")
+    val S3_BUCKET = sys.env("S3_BUCKET")
+    val S3_PREFIX = sys.env("S3_PREFIX")
```

Comment:
It would be nice to use the same config file for everything. Note that configuration can still be passed from outside, via `-Dmyapp.foo.bar=10`